packages/vite/src/node/logger.ts

```diff
@@ -129,7 +132,7 @@ export function printServerUrls(
         const type = detail.address.includes('127.0.0.1')
           ? 'Local:   '
           : 'Network: '
-        const host = detail.address
+        const host = detail.address.replace('127.0.0.1', hostname.name)
```

Comment:
This shouldn't be required anymore, right?

```suggestion
        const host = hostname.name
```

Reply:
This is still needed, when the user uses host `true`, `localhost`, `0.0.0.0` or `::`, then we need to show localhost in the console (same as what we do with --open)

Reply:
Ah, I see. But you're passing in `Hostname`, which already has that logic to resolve `name`, since it went through `resolveHostname`. Perhaps this is clearer (?):

```ts
.map((detail) =>
  detail.address.includes('127.0.0.1'))
    ? ` > Local:   ${chalk.cyan(`${protocol}://${hostname.name}:${chalk.bold(port)}${base}`)}`
    : ` > Network: ${chalk.cyan(`${protocol}://${detail.address}:${chalk.bold(port)}${base}`)}`,
)
```

Reply:
There is a bit of repetition in that option, maybe in another PR we can refactor and have `printURL` for this.